libs/template/file.go

```diff
@@ -97,5 +104,34 @@ func (f *inMemoryFile) PersistToDisk() error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(path, f.content, f.perm)
+
+	return writeFile(f.ctx, path, f.content, f.perm)
+}
+
+func shouldUseImportNotebook(ctx context.Context, path string, content []byte) bool {
+	if strings.HasPrefix(path, "/Workspace/") && runtime.RunsOnDatabricks(ctx) {
+		isNotebook, _, _ := notebook.DetectWithContent(path, content)
```

Comment:
We should return false instead of `isNotebook` if there's an error. Also please log the error. You can use `log.Debugf(ctx, ...)` to do so.